tests/draft2020-12/unevaluatedItems.json

```
@@ -485,5 +485,183 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "unevaluatedItems depends on adjacent contains",
+        "schema": {
+            "prefixItems": [true],
+            "contains": {"type": "string"},
+            "unevaluatedItems": false
+        },
+        "tests": [
+            {
+                "description": "second item is evaluated by contains",
+                "data": [ 1, "foo" ],
+                "valid": true
+            },
+            {
+                "description": "contains fails, second item is not evaluated",
+                "data": [ 1, 2 ],
+                "valid": false
+            },
+            {
+                "description": "contains passes, second item is not evaluated",
+                "data": [ 1, 2, "foo" ],
+                "valid": false
+            }
+        ]
+    },
```

I think the use of `prefixItems` is unnecessary and just complicates the tests. We already have tests that cover how `prefixItems` interacts with `unevaluatedItems`. We don't need that here too.

```suggestion
    {
        "description": "unevaluatedItems depends on adjacent contains",
        "schema": {
            "contains": {"type": "string"},
            "unevaluatedItems": false
        },
        "tests": [
            {
                "description": "first item is evaluated by contains",
                "data": [ "foo" ],
                "valid": true
            },
            {
                "description": "contains fails, first item is not evaluated",
                "data": [ 2 ],
                "valid": false
            },
            {
                "description": "contains passes, first item is not evaluated",
                "data": [ 2, "foo" ],
                "valid": false
            }
        ]
    },
```

---

`prefixItems` influences the interaction between `contains` and `unevaluatedItems`. An implementation must compare the indices produced by `contains` against those left over after evaluating `prefixItems`. If you leave out `prefixItems`, it makes it easier for implementations to pass, because the left over items are just the initial instance array.

My implementation is a case in point. If we remove `prefixItems`, my implementation would have passed these tests, but including `prefixItems` exposed a bug where I was not enumerating the left over indices correctly.

---

> If we remove `prefixItems`, my implementation would have passed these tests

That's a good enough reason for me. Let's keep it.